crates/core_simd/examples/matrix_inversion.rs

```diff
@@ -0,0 +1,316 @@
+//! 4x4 matrix inverse
+// Code ported from the `packed_simd` crate
+// Run this code with `cargo test --example matrix_inversion`
+#![feature(array_chunks)]
+use core_simd::*;
+
+// Gotta define our own 4x4 matrix since Rust doesn't ship multidim arrays yet :^)
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd)]
+pub struct Matrix4x4([[f32; 4]; 4]);
+
+#[allow(clippy::too_many_lines)]
+pub fn scalar_inv4x4(m: Matrix4x4) -> Option<Matrix4x4> {
+    let m = m.0;
+
+    let mut inv = [
+        [ // row 0:
+            // 0,0:
+            m[1][1]  * m[2][2] * m[3][3] -
```

Comment:
Some of these lines seem to have extra spaces for some reason?

Reply:
Yeah, `cargo fmt` actually is not _totally_ awful formatting for this, surprisingly, so I suggest that as a base.

Reply:
I think the function has the formatter turned off.

Reply:
We can address formatting concerns for the repo later.